artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3128,6 +3128,9 @@ private SlowConsumerReaperRunnable(long checkPeriod, float threshold, SlowConsum
 
       @Override
       public void run() {
+         if (getConsumerCount() == 0) {
```

Comment:
Can you remove this? this is a useless statement here.
you will be counting the number of elements on consumerSet once, for than later perform a for (Consumer consumer : getConsumerSet()) that will perform the very same operation. It is duplicating effort for no improvement IMO.


If you are having a race condition shown by a test, then it's a different issue that we must investigate, this would only hide the real issue if that was the case.